src/common/RoleIcon.tsx

```diff
@@ -3,8 +3,15 @@ import PropTypes from 'prop-types';
 
 import ROLES from 'game/ROLES';
 
-const RoleIcon = ({ id, className, ...others }) => {
-  let iconName;
+import Icon from './Icon';
+
+interface Props extends Omit<React.ComponentProps<typeof Icon>, 'id' | 'icon'> {
```

Comment:
Is there a reason to extend Icon here? The props from Icon are not used at all

Reply:
Fixed - thanks!